### PR TITLE
Implement get input dimension get output dimension in layernode

### PR DESCRIPTION
**Pull request by lhs8928**
implement getInputDimension, getOutputDimension in layer_node
    
    Self evaluation:
    
    Build test: [X]Passed [ ]Failed [ ]Skipped
    Run test: [ ]Passed [X]Failed [ ]Skipped

### CODE REVIEW

File: nntrainer/models/neuralnet.cpp [hunk 1]
```diff
@@ -446,8 +446,8 @@ void NeuralNetwork::setBatchSize(unsigned int batch) {
 
 bool NeuralNetwork::validateInput(sharedConstTensors X) {
 
-  auto &first_layer = model_graph.getSortedLayerNode(0)->getObject();
-  auto input_dim = first_layer->getInputDimension();
+  auto first_layer_node = model_graph.getSortedLayerNode(0);
```
Comment (kparichay) [inline nntrainer/models/neuralnet.cpp:449]: auto const &?

**Review by kparichay — APPROVED** [review]
LGTM other than the minor comment

**Review by zhoonit — APPROVED** [review]
LGTM :)

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: nntrainer/models/neuralnet.cpp (hunk 1 above)
Comment (lhs8928) [reply nntrainer/models/neuralnet.cpp]: Thanks

File: nntrainer/layers/layer_node.h [hunk 2]
```diff
@@ -247,6 +247,22 @@ class LayerNode : public ml::train::Layer, public GraphNode {
     layer->setNumOutputs(layers.size());
   }
 
+  /**
+   * @brief Get the input dimension
+   * @return TensorDim dimension of the input
+   */
+  const std::vector<TensorDim> &getInputDimensions() const {
+    return init_context.getInputDimensions();
```
[the remark this answers is not in the document]
Comment (lhs8928) [reply nntrainer/layers/layer_node.h]: Done

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**